src/colorizer/Dataset.ts

```diff
@@ -104,6 +105,7 @@ export default class Dataset {
     this.backdropLoader = frameLoader || new ImageFrameLoader(RGBAFormat);
     this.backdropData = new Map();
 
+    this.cleanupArrayLoaderOnDispose = !arrayLoader;
```

Comment:
This is smart and I like it! Reminds me a bit of some hacks I've seen around the neuroglancer codebase, which invents its own reference-counting types to ensure shared resources get cleaned up properly.